### PR TITLE
get more tests running for SQL

### PR DESCRIPTION
**Pull request by snopoke**
:blowfish: @benrudolph 

cc @calellowitz @biyeun

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/cleanup.py [hunk 1]
```diff
@@ -60,7 +61,7 @@ def rebuild_case_from_forms(domain, case_id, detail):
 
     return FormProcessorInterface(domain).hard_rebuild_case(case_id, detail)
 
-
+@unit_testing_only
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/cleanup.py]: E302 expected 2 blank lines, found 1

File: corehq/ex-submodules/casexml/apps/case/cleanup.py [hunk 2]
```diff
@@ -72,21 +73,12 @@ def safe_hard_delete(case):
     This is used primarily for cleaning up system cases/actions (e.g. the location delegate case).
     """
     if case.reverse_indices:
-        raise CommCareCaseError("You can't hard delete a case that has other dependencies ({})!".format(case._id))
-    forms = get_case_forms(case._id)
+        raise CommCareCaseError("You can't hard delete a case that has other dependencies ({})!".format(case.case_id))
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/cleanup.py]: E501 line too long (118 > 115 characters)

File: corehq/ex-submodules/casexml/apps/case/tests/test_delete.py [hunk 3]
```diff
@@ -1,26 +1,29 @@
...
-        cls.interface = FormProcessorInterface(TEST_DOMAIN_NAME)
+    def setUp(self):
+        self.interface = FormProcessorInterface(TEST_DOMAIN_NAME)
 
+    @run_with_all_backends
     def test_simple_delete(self):
         factory = CaseFactory()
         case = factory.create_case()
-        [case] = factory.create_or_update_case(CaseStructure(case_id=case._id, attrs={'update': {'foo': 'bar'}}))
+        [case] = factory.create_or_update_case(CaseStructure(case_id=case.case_id, attrs={'update': {'foo': 'bar'}}))
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/tests/test_delete.py]: E501 line too long (117 > 115 characters)

**Comment by benrudolph** [comment]
Looks good :+1: real failure:
```
======================================================================
ERROR: testComplicatedGatesBug (casexml.apps.phone.tests.test_sync_mode.MultiUserSyncTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 1325, in testComplicatedGatesBug
    form = self._postWithSyncToken(os.path.join(folder_path, f), self.sync_log.get_id)
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 79, in _postWithSyncToken
    _, form, _ = submit_form_locally(xml_data, last_sync_token=token_id)
TypeError: submit_form_locally() takes exactly 2 arguments (1 given)
```

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**